### PR TITLE
Add missing manifest_path field to Artifact

### LINKED ISSUE #187: Why isn't `manifest_path` made available?
I'm writing some code to parse `cargo test` output, and among the data given I want to grab the `manifest_path` from a `"reason":"compiler-artifact"` (i.e. `Message::CompilerArtifact(Artifact)`), but it's not there. The `Artifact` type has no `manifest_path`. It's there in the underlying JSON, so... how do I get to it? Here's an example output line:
```json
{"reason":"compiler-artifact","package_id":"bar 1.2.3 (path+file:///C:/Users/alexs/Documents/Projects/foo/bar)","manifest_path":"C:\\Users\\alexs\\Documents\\Projects\\foo\\bar\\Cargo.toml","target":{"kind":["lib"],"crate_types":["lib"],"name":"bar","src_path":"C:\\Users\\alexs\\Documents\\Projects\\foo\\bar\\lib.rs","edition":"2021","doc":true,"doctest":true,"test":true},"profile":{"opt_level":"1","debuginfo":2,"debug_assertions":true,"overflow_checks":true,"test":true},"features":[],"filenames":["C:\\Users\\alexs\\Documents\\Projects\\foo\\target\\debug\\deps\\bar-812df0f62c7b9322.exe","C:\\Users\\alexs\\Documents\\Projects\\foo\\target\\debug\\deps\\bar-812df0f62c7b9322.pdb"],"executable":"C:\\Users\\alexs\\Documents\\Projects\\foo\\target\\debug\\deps\\bar-812df0f62c7b9322.exe","fresh":true}
```
and output in `Message::parse_stream`:
```
Artifact {
    package_id: PackageId {
        repr: "bar 1.2.3 (path+file:///C:/Users/alexs/Documents/Projects/foo/bar)",
    },
    target: Target {
        name: "bar",
        kind: [
            "lib",
        ],
        crate_types: [
            "lib",
        ],
        required_features: [],
        src_path: "C:\\Users\\alexs\\Documents\\Projects\\foo\\bar\\lib.rs",
        edition: E2021,
        doctest: true,
        test: true,
        doc: true,
    },
    profile: ArtifactProfile {
        opt_level: "1",
        debuginfo: Some(
            2,
        ),
        debug_assertions: true,
        overflow_checks: true,
        test: true,
    },
    features: [],
    filenames: [
        "C:\\Users\\alexs\\Documents\\Projects\\foo\\target\\debug\\deps\\bar-812df0f62c7b9322.exe",
        "C:\\Users\\alexs\\Documents\\Projects\\foo\\target\\debug\\deps\\bar-812df0f62c7b9322.pdb",
    ],
    executable: Some(
        "C:\\Users\\alexs\\Documents\\Projects\\foo\\target\\debug\\deps\\bar-812df0f62c7b9322.exe",
    ),
    fresh: true,
}
```
Unless I'm missing something, it's the *only field* not present, too. Just my luck!

### PR DESCRIPTION
**Pull request by morr0ne**
Fixes #187

### CODE REVIEW

**Comment by morr0ne** [comment]
I did overlook that the test samples need to be fixed

**Comment by morr0ne** [comment]
Okay so apperantaly `parse_stream_is_robust` is failing but I'm not sure why.

**Comment by morr0ne** [comment]
Nevermind I figure it out, I also added the proper entry to the changelog.

File: src/messages.rs [hunk 1]
```diff
@@ -34,6 +34,8 @@ pub struct ArtifactProfile {
 pub struct Artifact {
     /// The package this artifact belongs to
     pub package_id: PackageId,
+    /// Path containing the `Cargo.toml`
```
Comment (oli-obk) [inline src/messages.rs:37]: I think it's the path to the Cargo.toml. the current doc comment sounds like it would point to the parent folder

Comment (morr0ne) [reply src/messages.rs:37]: Yeah good catch, I'll fix it.

**Pull request merged**